src/functionaltests/java/com/ericsson/ei/utils/DataBaseManager.java

```diff
@@ -138,16 +134,19 @@ public boolean verifyAggregatedObjectExistsInDB() throws InterruptedException {
      * @return list of missing events
      */
     private List<String> compareSentEventsWithEventsInDB(List<String> checklist) {
-        mongoClient = new MongoClient(getMongoDbHost(), getMongoDbPort());
-        MongoDatabase db = mongoClient.getDatabase(database);
-        MongoCollection<Document> table = db.getCollection(collection);
-        List<Document> documents = table.find().into(new ArrayList<>());
-        for (Document document : documents) {
-            for (String expectedID : new ArrayList<>(checklist)) {
-                if (expectedID.equals(document.get("_id").toString())) {
-                    checklist.remove(expectedID);
+        MongoCollection<Document> table = getCollection(eventObjectCollectionName);
+        try {
+            List<Document> documents = table.find().into(new ArrayList<>());
+
+            for (Document document : documents) {
```

Comment:
can't we use streams here?

Reply:
Streams are not recommended to use since they are slower than normal for loops and harder to read.

Reply:
After discussing with team it seems like majority things streams is harder to read. In Java 8 streams is slow, not sure if that is improved in later Java versions.